src/sentry/killswitches.py

```diff
@@ -279,7 +279,7 @@ def killswitch_matches_context(killswitch_name: str, context: Context, emit_metr
     option_value = options.get(killswitch_name)
     rv = _value_matches(killswitch_name, option_value, context)
 
-    if emit_metrics:
+    if emit_metrics and options.get("system.emit-kill-switch-metrics"):
```

Comment:
I didn't see any callers that set `emit_metrics=False`. If you wanted to make a smaller change, you could have the option drive the sampling rate of the metric instead of dropping the metric entirely.

Reply:
Yes, but what is the purpose of having this metric in the first place? If `killswitch` is not enabled (I assume for 99% of the cases it is not enabled), this metric will behave similar to `@metrics.wraps("ingest_consumer.process_event")`.